middleware/auth.go

```diff
@@ -0,0 +1,29 @@
+package middleware
+
+import (
+	"log"
+	"net/http"
+
+	"github.com/labstack/echo/v4"
+)
+
+const testToken = "gwi-token-12345" //for future implementation this token should be encrypted and handled properly
+
+func AuthMiddleware(next echo.HandlerFunc) echo.HandlerFunc {
```

Comment:
I see that you chose to use a test token here, but can you describe to me in simple words the steps of proper token handling??

Reply:
Of course! For the sake of simplicity and time management I have used a simple token-based authentication with a static token but in a real world application a JWT implementation is a more proper approach as it is safer and more reliable. So in order to implement a proper token handling we can use Golang-JWT package and we should go through the following steps:

- First of all we have to create the JWT Tokens. We also have to specify a signing method and any relevant info such as username and token expiration date.
- Then we have to sign the token with a secret key and return the generated token as a string
- Before granting access to protected resources we need to verify it’s authenticity. We provide a callback function to retrieve the secret key used for signing the token. If the token is valid, we continue processing the request; otherwise, we return an error indicating that the token is invalid
- To secure protected routes, we need to introduce a login system where users can authenticate and obtain a JWT token
- After implementing the login system and obtained a JWT token, we need to  modify the protected route to require a valid JWT token for access. In order to do this we can check for a valid JWT token in the Authorization header of the request. If the token is missing or invalid, we return an appropriate error response. Otherwise, we proceed with serving the protected resource.